htlcswitch/switch.go

```diff
@@ -297,6 +296,20 @@ type Switch struct {
 	// ack in the forwarding package of the outgoing link. This was added to
 	// make pipelining settles more efficient.
 	pendingSettleFails []channeldb.SettleFailRef
+
+	// aliasToReal is a map used for option_scid_alias links. The alias
+	// SCID is the key and the real, confirmed SCID is the value. This must
+	// be accessed with the indexMtx.
+	aliasToReal map[lnwire.ShortChannelID]lnwire.ShortChannelID
```

Comment:
Ah cool, was wondering how we planned on handling this. I like the approach of just yet-another-index, instead of needing to modify our existing ones or some some other type of translation.